### ISSUE TITLE
Keyboard Navigation Broken on List View Sidebar

### ISSUE THREAD

**Issue by jeryj** [WordPress/gutenberg#50639]
### Description

The keyboard navigation for the list view sidebar only works with Tab presses, but skips to[ the hidden navigation block](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/sidebar-navigation-screen-navigation-menus/navigation-menu-content.js#L194-L198) that is currently required for the navigation.

### Step-by-step reproduction instructions

1. Go to the Site Editor
2. Click Navigation from the left sidebar
3. Try to navigate the menu using your keyboard

### Screenshots, screen recording, code snippet

![Sidebar List View Navigation Keyboard Broken](https://github.com/WordPress/gutenberg/assets/967608/4657b218-9a40-4460-9c89-d74e52ab1508)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by draganescu** [comment]
How should it ideally work?

**Comment by jeryj** [comment]
I believe it's coded like a listbox, so the whole menu is one tab stop, and then you can use the up/down arrows to navigate between the menu items. I believe the navigation for the list view is fine when it's not bundled with a hidden block.

**Comment by draganescu** [comment]
I think we should stop using the LinkUI as it is an improper affordance for a sidebar and then we can just really hide  (not only visually) and then we can count of ListView to manage all keyboard navigation.

**Comment by getdave** [comment]
@jeryj This should be fixed now we've removed all editing from the sidebar.
